napari_allencell_annotator/widgets/template_item.py

```diff
@@ -115,6 +126,35 @@ def create_evt_listener(self):
             self.editable_widget.stateChanged.connect(lambda: self.parent.setCurrentItem(self))
         elif self._type == ItemType.LIST:
             self.editable_widget.activated.connect(lambda: self.parent.setCurrentItem(self))
+        elif self._type == ItemType.POINT:
+            self.editable_widget.clicked.connect(lambda: self.parent.setCurrentItem(self))
+            self.editable_widget.clicked.connect(self._handle_select_button_clicked)
+            self._annotation_model.annotation_started_changed.connect(self._handle_select_button_enabled)
+            self.parent.currentItemChanged.connect(self._toggle_button_off)
+
+    def _handle_select_button_enabled(self) -> None:
+        """
+        Enable and disable the Select button when annotation starts and stops. When the button is disabled, the text is set to Select.
+        """
+        self.editable_widget.setEnabled(self._annotation_model.is_annotation_started())
+
+        if not self.editable_widget.isEnabled():
+            self.editable_widget.setText("Select")
+
+    def _toggle_button_off(self) -> None:
+        """Change button texts to Select except for the current annotation item."""
+        if self != self.parent.currentItem():
+            self.editable_widget.setText("Select")
```

Comment:
try to rethink this function if possible:
avoid explicitly referencing `self`
avoid assigning signal to parent

Reply:
removed the if statement in https://github.com/aics-int/napari-allencell-annotator/pull/94/commits/753a22357cbab1d8f9a9b796cc7be0726c074af9

Reply:
can we avoid assigning signal to the parent here?

Reply:
can you expand more on that? I already removed the if statement?

Reply:
avoiding this: `self.parent.currentItemChanged.connect(self._toggle_button_off)`
for now we can ignore so we can get this merged in